tools-and-tests/suites/src/main/java/org/hiero/block/suites/e2e/BlockNodeAPITests.java

```diff
@@ -470,4 +472,85 @@ void swappedPublisherConnection() throws InterruptedException {
+        // change to List to allow multiple items
+        PublishStreamRequest request = PublishStreamRequest.newBuilder()
+                .blockItems(BlockItemSet.newBuilder().blockItems(blockItems).build())
+                .build();
+        CountDownLatch publishCountDownLatch = responseObserver.setAndGetOnNextLatch(1);
+        requestStream.onNext(request);
+        endBlock(blockNumber, requestStream);
+
+        publishCountDownLatch.await(); // wait for acknowledgement response
+        assertThat(responseObserver.getOnNextCalls())
+                .hasSize(1)
+                .first()
+                .returns(PublishStreamResponse.ResponseOneOfType.ACKNOWLEDGEMENT, responseKindExtractor)
+                .returns(0L, acknowledgementBlockNumberExtractor);
+
+        // Assert no other responses sent
+        assertThat(responseObserver.getOnErrorCalls()).isEmpty();
+        assertThat(responseObserver.getOnSubscriptionCalls()).isEmpty();
+        assertThat(responseObserver.getOnCompleteCalls().get()).isEqualTo(0);
+        assertThat(responseObserver.getClientEndStreamCalls().get()).isEqualTo(0);
+
+        // ==== Scenario 2: Publish duplicate genesis block and confirm duplicate block response and stream closure ===
+        CountDownLatch publishCompleteCountDownLatch = responseObserver.setAndGetOnCompleteLatch(1);
+        requestStream.onNext(request);
+        endBlock(blockNumber, requestStream);
+
+        publishCompleteCountDownLatch.await(); // wait for onComplete caused by duplicate response
+
+        // Assert that one more response is sent.
+        assertThat(responseObserver.getOnNextCalls())
+                .hasSize(2)
+                .element(1)
+                .returns(PublishStreamResponse.ResponseOneOfType.END_STREAM, responseKindExtractor)
+                .returns(PublishStreamResponse.EndOfStream.Code.DUPLICATE_BLOCK, endStreamResponseCodeExtractor)
+                .returns(0L, endStreamBlockNumberExtractor);
+
+        // Assert no other responses sent
+        assertThat(responseObserver.getOnErrorCalls()).isEmpty();
+        assertThat(responseObserver.getOnSubscriptionCalls()).isEmpty();
+        assertThat(responseObserver.getOnCompleteCalls().get()).isEqualTo(1);
+        assertThat(responseObserver.getClientEndStreamCalls().get()).isEqualTo(0);
+
+        final long blockNumber1 = 1;
+        BlockItem[] blockItems1 = BlockItemBuilderUtils.createSimpleBlockWithNumber(blockNumber1);
+        PublishStreamRequest request1 = PublishStreamRequest.newBuilder()
+                .blockItems(BlockItemSet.newBuilder().blockItems(blockItems1).build())
+                .build();
+
+        // This asserts that UncheckedIOException is thrown and its cause is a SocketException with "socket closed"
+        UncheckedIOException ex = assertThrows(UncheckedIOException.class, () -> {
+            requestStream.onNext(request1);
```

Comment:
It's not clear where you expect the exception to be thrown? If it happened on any line the check would still pass right?

Reply:
It will be thrown on the first `onNext()`. Added the note for clarity